docs/upgrade.html

```diff
@@ -17,7 +17,56 @@
 
 <script><!--#include virtual="js/templateData.js" --></script>
 
-<script id="upgrade-template" type="text/x-handlebars-template">
+<h4><a id="upgrade_11_1_0" href="#upgrade_11_1_0">Upgrading from 0.8.x, 0.9.x, 0.10.0.x, 0.10.1.x, 0.10.2.x or 0.11.0.0 to 0.11.1.0</a></h4>
+<p>Kafka 0.11.1.0 introduces wire protocol changes. By following the recommended rolling upgrade plan below,
+    you guarantee no downtime during the upgrade. However, please review the <a href="#upgrade_1110_notable">notable changes in 0.11.1.0</a> before upgrading.
+</p>
+
+<p><b>For a rolling upgrade:</b></p>
+
+<ol>
+    <li> Update server.properties on all brokers and add the following properties. CURRENT_KAFKA_VERSION refers to the version you
+        are upgrading from. CURRENT_MESSAGE_FORMAT_VERSION refers to the current message format version currently in use. If you have
+        not overridden the message format previously, then CURRENT_MESSAGE_FORMAT_VERSION should be set to match CURRENT_KAFKA_VERSION.
+        <ul>
+            <li>inter.broker.protocol.version=CURRENT_KAFKA_VERSION (e.g. 0.8.2, 0.9.0, 0.10.0, 0.10.1, 0.10.2, 0.11.0).</li>
+        </ul>
+    </li>
+    <li> Upgrade the brokers one at a time: shut down the broker, update the code, and restart it. </li>
+    <li> Once the entire cluster is upgraded, bump the protocol version by editing <code>inter.broker.protocol.version</code> and setting it to 0.11.1.
+    <li> Restart the brokers one by one for the new protocol version to take effect. </li>
+</ol>
+
+<p><b>Additional Upgrade Notes:</b></p>
+
+<ol>
+    <li>If you are willing to accept downtime, you can simply take all the brokers down, update the code and start them back up. They will start
+        with the new protocol by default.</li>
+    <li>Bumping the protocol version and restarting can be done any time after the brokers are upgraded. It does not have to be immediately after.
+        Similarly for the message format version.</li>
+</ol>
+
+<h5><a id="upgrade_1110_notable" href="#upgrade_1100_notable">Notable changes in 0.11.1.0</a></h5>
+<ul>
+    <li>A broker will now stay online to serve replicas on live log directories even if there are offline log directories. A log
+        directory may become offline due to IOException caused by hardware failure. Users need to monitor the per-broker metric
+        <code>offlineLogDirectoryCount</code> to check whether is offline log directory. </li>
+    <li>If a new replica is assigned to be created on a broker with offline log directory AND the
+        <code>inter.broker.protocol.version</code> is 0.11.0 or earlier, the broker will not create this replica and the replica will
+        be offline until the log failure on that broker is fixed. In order to make sure that new topics not be created with offline replicas,
+        user can use upgrade <code>inter.broker.protocol.version</code> to 0.11.1 or simply shutdown the broker before the hardware issue is fixed.</li>
```

Comment:
As long as the controller and the broker is running new code, the controller will be able to handle the failed disk dir event even if the inter-broker protocol is still old. More specifically, if there is log directory failure, the broker will write a sequential znode in zookeeper, the controller will send LeaderAndIsrRequest, broker will specify error in the response, and the controller will trigger leader election. None of these steps require new inter-broker protocol. Does this make sense?

I am not sure we should add a config just to take care of the potential bug. Ideally we don't want to add a config just for short-term use. The typical way of handling bug is to either hotfix the code or rollback to the last stable version. Any KIP or big code change may potentially have bug and it seems a big weird to add a config to handle the bug.

Adding this config is easy since all we need is to tell ReplicaManager not to start that LogDirFailureHandler. Do you think this is necessary?